docs/ADR-4-Era-selectors-for-CLI-commands.md

```diff
@@ -0,0 +1,97 @@
+
+Would it make sense for "magical" commands like the `transaction build` command to try to be more helpful and use the current era as the default?  That would _seem_ to be in the spirit of the command.
+
+Historically we changed the default transactions era in the first release after the hard fork in order to give the tool/dapp developers the possibility to update the default transaction era in their code after the hard fork - otherwise their tools would stop working after the hard fork.
+
+Care needs to be taken to avoid making a CLI release that does not work by default by defaulting to using an era that is not yet active. So that's why we have historically only updated the default era in the release after the hard fork, not the release before.
+
+This way, third party tools that use the CLI would not stop working after the hard fork, they would continue to create and submit transactions in the older format, for the older era. Historically we always made that backwards compatibility work. It just meant that CLI users would not get the new features for the new era without explicitly using the right era flag - until the following release changed the default.
+
+Moreover, it is often the case that CLI commands in the release that crossed the hard fork may not yet be updated to work in the new era.  If we adopted policy of using the current era for our CLI commands, even "magical" commands that try to infer as much as possible, we risk possibility of the command breaking after the hard fork.
+
+This was the context from the adoption of [[ADR-1 Default eras for CLI commands]] until now.
+
+Going foward we have additional concerns to consider:
+
+Some commands require access to the protocol parameters in the era the command is operating in and the contents of the protocol parameters could have meaningful bearing on how any transactions derived from those commands will be interpreted on-chain.
+
+It is important therefore that the protocol parameters that is used contain the sorts of value that reflect the current state of the ledger.
+
+Unfortunately, the upcoming Conway era introduces many new protocol parameters and obsoletes many others, meaning that there is no way to construct sensible values for these protocol parameters that would guarantee that transactions derived from them operate in a predictable way on chain.
+
+This would mean that submitting transactions built for an older era inherently carries with the risk that the transaction behaves other than intended.
+
+This is especially true for transactions built in an older era with an incomplete, made up or out-of-date values.
+
+This is also true for transactions upgraded from an older era, although to a lesser extent.
+
+Constructing a command in the currently era is the for any command that requires protocol-parametes is the only way to guarantee a transaction works as intended.
```

Comment:
This reads weird :slightly_frowning_face: Maybe a mash-up of two different versions of the sentence?